### ISSUE TITLE
Estimation of parent nodes when passing from tips to root

### ISSUE THREAD

**Issue by MichaelSev** [ComputationalEvolutionaryMorphometry/hyperiax#56]
When not using the; 
upmodel = UpLambdaReducer(up, transform, 
                   reductions={
                       'weighted_value': 'sum', 
                       'inverse_edge_length': 'sum'
                    }
        )
        
the indexing of computational nodes are flawed, and inner nodes are for computation, before their children have been estimated for some types of topologies. 

The problem arise, when this LambdaReducer are not used, in the case of functions which does not only rely on the "sum", or "prod", for estimating of values to parent nodes. 

A quick fix is to mask the premature estimates in the estimated function, but this is not sustainable for long term 
result = jnp.where(leaf_mask[:, None], value, result)

**Comment by MichaelSev** [comment]
Solved in #63
